### PR TITLE
Always populate and prune events

### LINKED ISSUE #4813: Unconditionally populate/prune events data irrespective of experimental cache flag value
The experimental event-based cache flag currently gates population/pruning of the events table. Unfortunately this means that you cannot have a mixed deployment of servers with new servers where the feature is enabled and those where it is not (or old servers that don't have the feature at all). The consequence of mixing servers is that flag-enabled servers do not observe changes to entries/agents made by other servers, causing an incorrect authorized entry view for agents connected to the flag-enabled servers.

This breaks rolling upgrade of servers as well as mixed deployment testing.

The former is particularly heinous when considering the upgrade between 1.9.0 and (hopefully) 1.10.0 when this feature is enabled by default.

As such, the 1.9.0 release should unconditionally populate and prune the events data, irrespective of whether or not the events-based cache is utilized.

### PR DESCRIPTION
**Pull request by faisal-memon**
**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**

Events table

**Description of change**

Events will always be populated and pruned regardless of events based cache being used.

**Which issue this PR fixes**

fixes #4813

### CODE REVIEW

File: pkg/server/endpoints/endpoints_test.go
Comment (rturner3) [inline pkg/server/endpoints/endpoints_test.go:1]: 1. Can you add an assertion into `TestNew()` to make sure we're wiring up the prune task properly into the `Endpoints` struct?
```go
assert.NotNil(t, endpoints.EntryFetcherPruneEventsTask)
```
2. Can you also wire up the entry fetcher prune task into the `Endpoints` struct constructed in `TestListenAndServe()` to more closely resemble how the struct is expected to be built?

**Review by rturner3 — COMMENTED** [review]
Changes are looking good. I left a couple small comments on how we can unit test them

File: pkg/server/endpoints/endpoints_test.go
Comment (faisal-memon) [reply pkg/server/endpoints/endpoints_test.go:1]: done

**Review by rturner3 — DISMISSED** [review]

File: pkg/server/datastore/sqlstore/sqlstore.go [hunk 1]
```diff
@@ -265,7 +265,7 @@ func (ds *Plugin) CreateAttestedNode(ctx context.Context, node *common.AttestedN
 		if err != nil {
 			return err
 		}
-		return createAttestedNodeEvent(tx, node.SpiffeId, ds.eventsBasedCache)
```
[the remark this answers is not in the document]
Comment (faisal-memon) [reply pkg/server/datastore/sqlstore/sqlstore.go:268]: It only gates the list events function now. So can be removed.

[the remark this answers is not in the document]
Comment (faisal-memon) [reply pkg/server/datastore/sqlstore/sqlstore.go:268]: @azdagron Would you like me to remove?

[the remark this answers is not in the document]
Comment (azdagron) [reply pkg/server/datastore/sqlstore/sqlstore.go:268]: I don't think there is risk one way or the other. If you have a few spare cycles, might be nice to just get it out of the way now. If not, no worries.

[the remark this answers is not in the document]
Comment (faisal-memon) [reply pkg/server/datastore/sqlstore/sqlstore.go:268]: Ok removed the flag

**Review by rturner3 — APPROVED** [review]

**Review by azdagron — APPROVED** [review]

**Pull request merged**